docs/tutorials/go/audiobook/index.md

```diff
@@ -0,0 +1,1026 @@
+Activities add check-in points for your application state in your [Event History](https://docs.temporal.io/workflows#event-history).
+When retried, an Activity picks up state from just before it was first called, so it's like the failed attempt never happened.
+
+The `TTSActivities` interface handles text-to-speech operations with the following methods:
+
+- **`ReadFile`**: Reads from the specified `fileInputPath` and returns the contents as a chunked array of strings, for well-sized API calls.
+- **`CreateTemporaryFile`**: Creates a temporary file for audio output in a designated folder that is typically cleaned up on system reboots.
+- **`Process`**: Sends a text chunk to OpenAI for processing, retrieves the TTS audio segment, and appends it to an output file.
+- **`MoveOutputFileToPlace`**: Moves an audio file to a safe, versioned location in the same folder as the original text file.
+
+Here is how the activities help in the overall conversion process:
+![After reading a text file and dividing it into chunks, each chunk is sent to OpenAI to be converted to audio and the results appended to the output file](images/highlevelprocess.png)
+
+### Prepare your text with `ReadFile`
+
+The `ReadFile` Activity reads and processes text from a file.
+Before and during the read, it performs a series of safety checks, such as making sure the file exists and it's readable.
+If it encounters any file system issues, it returns an application error.
+A corrupt file system can't be reasonably retried.
+
+The `ReadFile` activity reads and processes text from a file.
+It performs safety checks to ensure the file exists and is readable.
+If file system issues are encountered, it returns an error.
+Errors related to file system issues indicate fundamental issues that can't be resolved through retries.
+
+The following code appears at the end of the Activity.
+It splits the text into "chunks", each containing part of the source material:
+
+```go
+const (
+    apiEndpoint          = "https://api.openai.com/v1/audio/speech"
+    contentType          = "application/json"
+    requestTimeout       = 30 * time.Second
+    maxTokens            = 512
+    averageTokensPerWord = 1.33
+)
+
+...
+
+trimmedContent := strings.TrimSpace(string(content))
+words := strings.Fields(trimmedContent)
+chunks := []string{}
+chunk := strings.Builder{}
+
+for _, word := range words {
+    if float64(chunk.Len()+len(word))*averageTokensPerWord <= maxTokens {
+        if chunk.Len() > 0 {
+            chunk.WriteString(" ")
+        }
+        chunk.WriteString(word)
+    } else {
+        chunks = append(chunks, chunk.String())
+        chunk.Reset()
+        chunk.WriteString(word)
+    }
+}
+
+if chunk.Len() > 0 {
+    chunks = append(chunks, chunk.String())
+}
+
+return chunks, nil
+
+```
+
+As a rule-of-thumb, a typical English word uses 1.33 OpenAI tokens, which is how this code defines the `AVERAGE_TOKENS_PER_WORD` constant.
```

Comment:
I would swap this paragraph and the following one for clarity

Reply:
swapped